### PR TITLE
Deprecate squeeze(A, dims) to use keyword arguments

### PR DESCRIPTION
**Pull request by mbauman**
This is a follow-up to issue #25501 and seems to make sense in that same vein.  This also adds NEWS for squeeze and all the other related functions (#25501 was missing NEWS).

### CODE REVIEW

File: base/abstractarraymath.jl [hunk 1]
```diff
@@ -57,14 +57,15 @@ julia> a = reshape(Vector(1:4),(2,2,1,1))
...
-julia> squeeze(a,3)
+julia> squeeze(a; dims=3)
 2×2×1 Array{Int64,3}:
 [:, :, 1] =
  1  3
  2  4
 ```
 """
-function squeeze(A::AbstractArray, dims::Dims)
+squeeze(A; dims=throw(ArgumentError("the dimensions to squeeze must be specified with a `dims` keyword argument"))) = _squeeze(A, dims)
```
Comment (ararslan) [inline base/abstractarraymath.jl]: You should be able to get by with a required keyword argument here:
```julia
squeeze(A; dims) = _squeeze(A, dims)
```
That will throw an `UndefKeywordError` when `dims` is not provided.

Comment (mbauman) [reply base/abstractarraymath.jl]: Oh, right, I forgot we do this automatically now. Thanks!

**Review by ararslan — APPROVED** [review]

**Comment by ararslan** [comment]
64-bit Windows failure is a timeout.

**Comment by staticfloat** [comment]
Wow, apparently kwarg codegen has significantly improved since 0.6.  Nice!

**Pull request merged**